app/src/main/java/com/pokedex/mdb/pokedex/ListActivity.java

```diff
@@ -0,0 +1,230 @@
+        import android.support.v7.widget.RecyclerView;
+        import android.support.v7.widget.SearchView;
+        import android.view.Menu;
+        import android.view.MenuInflater;
+        import android.view.MenuItem;
+
+        import java.util.ArrayList;
+
+public class ListActivity extends AppCompatActivity {
+
+    private RecyclerView recycleboy;
+    private boolean isLinear = false;
+    private Pokedapter pokedapter;
+    private ArrayList<Pokedex.Pokemon> pokemons;
+    private ArrayList<Pokedex.Pokemon> searchingcache;
+    private ArrayList<Pokedex.Pokemon> filteringcache;
+
+    private ArrayList<Pokedex.Pokemon> lowHP;
+    private ArrayList<Pokedex.Pokemon> medHP;
+    private ArrayList<Pokedex.Pokemon> highHP;
+
+    private boolean lowSelected = true;
+    private boolean medSelected = true;
+    private boolean highSelected = true;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list);
+        recycleboy = (RecyclerView) findViewById(R.id.recycleboy);
+        Pokedex pokedex = new Pokedex();
+        pokemons = pokedex.getPokemon();
+        setFilter(pokemons);
+        setSearch(pokemons);
+        lowHP = new ArrayList<>();
+        medHP = new ArrayList<>();
+        highHP = new ArrayList<>();
+        updateRecyclerView(pokemons);
+
+        // filter low HP pokemon
+        for (int i = 0; i < pokemons.size(); i += 1) {
+            String hpString = pokemons.get(i).hp;
+            int hp = Integer.parseInt(hpString);
+            if (hp < 50) {
+                lowHP.add(pokemons.get(i));
+            }
+        }
+
+        // filter med HP pokemon
+        for (int i = 0; i < pokemons.size(); i += 1) {
+            String hpString = pokemons.get(i).hp;
+            int hp = Integer.parseInt(hpString);
+            if (hp >= 50 && hp <= 100) {
+                medHP.add(pokemons.get(i));
+            }
+        }
+
+        // filter high HP pokemon
+        for (int i = 0; i < pokemons.size(); i += 1) {
+            String hpString = pokemons.get(i).hp;
+            int hp = Integer.parseInt(hpString);
+            if (hp > 100) {
+                highHP.add(pokemons.get(i));
+            }
+        }
+    }
+
+    public ArrayList<Pokedex.Pokemon> getOverlap() {
+        ArrayList<Pokedex.Pokemon> p = new ArrayList<Pokedex.Pokemon>();
+        ArrayList<Pokedex.Pokemon> smaller = searchingcache;
+        ArrayList<Pokedex.Pokemon> longer = filteringcache;
+        if (smaller.size() > longer.size()) {
+            smaller = filteringcache;
+            longer = searchingcache;
+        }
+        for(int i = 0; i < smaller.size(); i++) {
+            if(longer.contains(smaller.get(i))) {
+                p.add(smaller.get(i));
+            }
+        }
+        return p;
+    }
+
+    public ArrayList<Pokedex.Pokemon> setSearch(ArrayList<Pokedex.Pokemon> p) {

```

Comment:
setSearch and setFilter are basically exactly the same, could make them one function that takes another ArrayList parameter